internal/controller/warehouses/images.go

```diff
@@ -129,11 +129,20 @@ func imageDiscoveryLogFields(sub kargoapi.ImageSubscription) []any {
 		"platformConstrained", sub.Platform != "",
 	}
 	switch sub.ImageSelectionStrategy {
-	case kargoapi.ImageSelectionStrategySemVer, kargoapi.ImageSelectionStrategyDigest:
-		f = append(
-			f,
-			"semverConstraint", sub.SemverConstraint,
-		)
+	case kargoapi.ImageSelectionStrategySemVer:
+		if sub.Constraint != "" {
+			f = append(
+				f,
+				"constraint", sub.Constraint)
+		} else if sub.SemverConstraint != "" {
+			f = append(
+				f,
+				"semverConstraint", sub.SemverConstraint)
+		}
+	case kargoapi.ImageSelectionStrategyDigest:
+		if sub.Constraint != "" {
+			f = append(f, "constraint", sub.Constraint)
+		}
```

Comment:
Similar to [this comment](https://github.com/akuity/kargo/pull/4541/files#r2185696503), this wasn't conditional before and I think there's a benefit to continuing to explicitly show in logs when the strategy is unconstrained.

```suggestion
		if sub.Constraint != "" {
			f = append(f, "constraint", sub.Constraint)
		}
```